lib/utils.js

```diff
@@ -304,4 +310,32 @@ utils.mapHeaders = function mapHeaders(headers, addHeaders) {
     return headers;
 };
 
+utils.isUnsupportedQuery = function isUnsupportedQuery(queryStr) {
+    if (!queryStr) {
+        return false;
+    }
+    // Reject unsupported query fields (e.g. ?accelerate), but allow
+    // the same strings as values (e.g. ?test=accelerate) and queries like
+    // '?acceleratefoobar' or '&foobaraccelerate'
+    return constants.unsupportedQueries.some(q => {
```

Comment:
This is the reasoning I was going with.
constants.unsupportedQueries is definitive (as in we control it's length, hence we control the loop length).
`String. startsWith`, `String.charAt`, `String.endsWith` access the string at locations as if its an array, so that shouldn't be of any any concern. `String.includes`, this I am curious but I don't think it will be a huge deal, as it would be an equivalent of `indexOf`. So overall this approach should be ok IMO.

The other way, `Object.keys(query)`, now if we get a request with 400 query params in the url (for example), it would be an expensive operation to extract the keys and loop through them, we are giving control of the loop length to the user which is prone to attacks.